### PR TITLE
Address Detailed Water Cooling Coil Sizing Problem

### LINKED ISSUE #8877: Detailed water cooling coil sizing problem when used in Unitary System placed on condenser loop
Issue overview
--------------
Testing shows that the detailed water cooling coil in UnitarySystem connected to condenser loop has air flow sizing problem. The design air vol flow rate is sized to zero which leads other detailed coil sizing parameters to zero. See error message below:

```
   ** Warning ** The design air flow rate is zero for Coil:Cooling:Water FREE COOLING COIL
   **   ~~~   ** The autosize value for max air volume flow rate is zero
   ** Severe  ** Coil:Cooling:Water:DetailedGeometry: "FREE COOLING COIL"
   **   ~~~   ** Coil Minimum Airflow Area must be greater than 0. Coil area = 0.000000
   **  Fatal  ** Preceding water coil sizing errors cause program termination
   ...Summary of Errors that led to program termination:
   ..... Reference severe error count=1
   ..... Last severe error=Coil:Cooling:Water:DetailedGeometry: "FREE COOLING COIL"
   ************* Warning:  Node connection errors not checked - most system input has not been read (see previous warning).
   ************* Fatal error -- final processing.  Program exited before simulations began.  See previous error messages.
   ************* EnergyPlus Warmup Error Summary. During Warmup: 0 Warning; 0 Severe Errors.
   ************* EnergyPlus Sizing Error Summary. During Sizing: 0 Warning; 0 Severe Errors.
   ************* EnergyPlus Terminated--Fatal Error Detected. 3 Warning; 1 Severe Errors; Elapsed Time=00hr 00min  4.38sec
```

Also note the first line of the warning message says Coil:Cooling:Water, instead of  Coil:Cooling:Water:DetailedGeometry because it is hard coded.
 
### Details
Some additional details for this issue (if relevant):
 - Develop(if using an intermediate build, include SHA)

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by Nigusse**
Pull request overview
---------------------
 - Fixes #8877 
 - Coil:Cooling:Water:DetailedGeometry in wrapped in unitarysystem and connected to demand side of a condenser loop fails during sizing

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by Nigusse** [comment]
@rraustad @mjwitte This fix allows the detailed water cooling coil sizing problem when the coil is placed in UnitarySystem and the coil is connected to a condenser loop demand side. With this change the CoilSystemWater cooling coil and the UnitarySystem runs successfully but may throw warning error depending on the what value is used for condenser loop exit temp.

File: src/EnergyPlus/WaterCoils.cc [hunk 1]
```diff
@@ -2364,6 +2364,11 @@ void SizeWaterCoil(EnergyPlusData &state, int const CoilNum)
                 TempSize = AutoSize; // get the autosized air volume flow rate for use in other calculations
             }
 
+            if (state.dataWaterCoils->WaterCoil(CoilNum).WaterCoilModel == iCoilModel::CoolingDetailed) { // Coil:Cooling:Water:DetailedGeometry
+                TempSize = AutoSize; // get the autosized air volume flow rate for use in other calculations
+                state.dataWaterCoils->WaterCoil(CoilNum).DesAirVolFlowRate = AutoSize;
+            }
+
             bool errorsFound = false;
```
Comment (Nigusse) [inline src/EnergyPlus/WaterCoils.cc:2373]: if state.dataWaterCoils->WaterCoil().DesAirVolFlowRate is zero, which is the case for this coil type, leads to zero value for WaterCoil(CoilNum).InletAirMassFlowRate. As the result other detailed cooling coil sizing parameters such minimum Coil Area and Fin Area will be sized to zero and the sizing calculation fatal out.  So, needed to autosize it but not report it. May be I need to set it to autosize if the value of WaterCoil().DesAirVolFlowRate is zero. I will make changes after reviewing the regression results.

File: src/EnergyPlus/WaterCoils.cc [hunk 2]
```diff
@@ -2559,7 +2564,7 @@ void SizeWaterCoil(EnergyPlusData &state, int const CoilNum)
 
             if (state.dataWaterCoils->WaterCoil(CoilNum).DesAirVolFlowRate <= 0.0) {
                 state.dataWaterCoils->WaterCoil(CoilNum).DesAirVolFlowRate = 0.0;
-                ShowWarningError(state, "The design air flow rate is zero for Coil:Cooling:Water " + state.dataWaterCoils->WaterCoil(CoilNum).Name);
+                ShowWarningError(state, "The design air flow rate is zero for " + CompType + " " + state.dataWaterCoils->WaterCoil(CoilNum).Name);
                 ShowContinueError(state, "The autosize value for max air volume flow rate is zero");
```
Comment (Nigusse) [inline src/EnergyPlus/WaterCoils.cc:2569]: Replaced hard wired coil type.

**Comment by Nigusse** [comment]
Added defect files.

**Comment by Nigusse** [comment]
Before the fix the defect files fail with the following severe error message:

```
   ** Warning ** The design air flow rate is zero for Coil:Cooling:Water FREE COOLING COIL
   **   ~~~   ** The autosize value for max air volume flow rate is zero
   ** Severe  ** Coil:Cooling:Water:DetailedGeometry: "FREE COOLING COIL"
   **   ~~~   ** Coil Minimum Airflow Area must be greater than 0. Coil area = 0.000000
   **  Fatal  ** Preceding water coil sizing errors cause program termination
   ...Summary of Errors that led to program termination:
   ..... Reference severe error count=1
   ..... Last severe error=Coil:Cooling:Water:DetailedGeometry: "FREE COOLING COIL"
```

**Comment by Nigusse** [comment]
The regression test suite results shows warning message in one example file (5ZoneCoolBeam.idf) only. This idf has detailed water cooling coil object. The diffs are in the eio file are during warmup periods only. And they are tiny and improved convergence.

**Comment by Myoldmopar** [comment]
@rraustad do you have any thoughts on this?  It appears to be reasonable and complete, but it would be good for you to at least take a peek and comment if you can.

**Comment by Myoldmopar** [comment]
Closing as it appears this is not needed per @Nigusse

**Pull request closed without merging**